### ISSUE TITLE
Cannot login to exchange server

### ISSUE THREAD

**Issue by jwcrawley** [nmarus/node-ews#26]
Greetings,

When I put in my credentials in the variables for this service, and then attempt to run the script, I get a NTLMStatusCode: 401 Unauthorized.

If I instead go to https://[SERVER]/EWS/Exchange.asmx in the browser , it prompts me for a login/password. It accepts my credentials, and then sends me to a page with a link to a .wsdl, the text "You have created a service." and some C#/VB code.

Is this related to the plain text issue others have been having? Or is this me missing something blindingly obvious?

Sincerely,
Josh

**Comment by nmarus** [comment]
unauthorized is typically related to the format of the username. I am using <username> for the exchange server I am connecting to. A few things that might be the cause:

* EWS API is case sensitive for username - You may want to check the case your account name has in AD. This is often a problem as the web interfaces and windows login, etc are not case-sensitive.
* Depending on the exchange server, you may need to use <nt-domain>\<username> or <username>@<ip-domain-name>

**Comment by aaroncalderon** [comment]
I am now getting the same error as @jwcrawley. It is odd, since it was working yesterday, but its not today.

I was using the form of  `[username]@[domain-name]` successfully. 

Since I started getting the error I have been using all  kinds of forms with no avail.

May have the system admins changed something that is causing my account to get denied access to the exchange services?

**Comment by eino-makitalo** [comment]
I have this very same issue with Exchange 2007 SP3... no ntlm login what ever I try. Does this support NTLMv2?

**Comment by eino-makitalo** [comment]
I got it working with
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
and then
````js
var username = 'user@example.local';
var password = 'xxxx';
var host = 'https://localhost';
````

**Comment by eino-makitalo** [comment]
To clarify my problem. I had a tunnel with ssh  from localhost to Exchange Server 2007 SP3 and I found out that it gave all the time 401 Unathorize when I had also options in use
````javascript
var options = {
 **rejectUnauthorized: false,**
 strictSSL: false
};
````
If you want to use options, take away rejectUnauthorized. I removed 4th parameter options away and found out that `process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';` was enough.  If I used rejectUnauthorized (true or false) I got 401 Unauthorized error all the time

I think this is because NTLM negotiation send 401 in the beginning and after type1 message. So it should continue to work.  Code seems to set rejectUnauhtorized=true before type3 message which is OK.  Before that, it's normal in protocol to get 401 error.
